eva-tools/src/main/java/embl/ebi/variation/eva/vcfdump/VariantExporter.java

```diff
@@ -316,69 +90,24 @@ private int getVariantStart(QueryOptions query) {
-        regionQuery.put("region", region.toString());
-        return regionQuery;
+        return variantsToExportByStudy;
     }
 
-    private void writeChunkToOutputFiles(Map<String, List<VariantContext>> variantsPerStudyInChunk, Map<String, VariantContextWriter> writers) {
-        for (Map.Entry<String, List<VariantContext>> variantsPerStudy : variantsPerStudyInChunk.entrySet()) {
-            String study = variantsPerStudy.getKey();
-            if (writers.containsKey(study)) {
-                VariantContextWriter writer = writers.get(study);
-                variantsPerStudy.getValue().forEach(writer::add);
-            }
-        }
-    }
-
-    private void dumpVariants(Map<String, VariantSource> sources, Map<String, VariantContextWriter> writers, QueryOptions query) {
-        query = addProjectionToQuery(query);
-        VariantDBIterator variantDBIterator = variantDBAdaptor.iterator(query);
-        while (variantDBIterator.hasNext()) {
-            Variant variant = variantDBIterator.next();
-            try {
-                Map<String, VariantContext> variantContexts = convertBiodataVariantToVariantContext(variant, sources, null);
-                for (Map.Entry<String, VariantContext> variantContextEntry : variantContexts.entrySet()) {
-                    if (writers.containsKey(variantContextEntry.getKey())) {
-                        writers.get(variantContextEntry.getKey()).add(variantContextEntry.getValue());
-                    }
-                }
-            } catch (Exception e) {
-                logger.info(String.format("Variant dump failed: \"%s:%d:%s>%s\"", variant.getChromosome(),
-                        variant.getStart(), variant.getReference(), variant.getAlternate()),
-                        e);
-                failedVariants++;
-            }
-            writtenVariants++;
-            if (writtenVariants % 100000 == 0) {
-                logger.info("written variants: " + writtenVariants);
-            }
+    public Map<String, VariantSource> getSources(VariantSourceDBAdaptor sourceDBAdaptor, List<String> studyIds) {
+        Map<String, VariantSource> sources = new TreeMap<>();
+        List<VariantSource> sourcesList = sourceDBAdaptor.getAllSourcesByStudyIds(studyIds, new QueryOptions()).getResult();
+        for (VariantSource variantSource : sourcesList) {
+            sources.put(variantSource.getStudyId(), variantSource);
         }
+        return sources;
     }
 
     /**
      * postconditions:
      * - returns one header per study (one header for each key in `sources`).
      * @throws IOException
      */
-    private Map<String, VCFHeader> getVcfHeaders(Map<String, VariantSource> sources) throws IOException {
+    public Map<String, VCFHeader> getVcfHeaders(Map<String, VariantSource> sources) throws IOException {
         Map<String, VCFHeader> headers = new TreeMap<>();
 
         for (VariantSource source : sources.values()) {
```

Comment:
I checked the PASS metadata description in the 1kGP3 dataset and the description string is "All filters passed". I suggest using that one because that makes clearer the meaning of "PASS".